service/matching/tasklist/matcher.go

```diff
@@ -123,16 +123,18 @@ func (tm *TaskMatcher) Offer(ctx context.Context, task *InternalTask) (bool, err
 		}
 	}
 
+	childCtx, cancel := newChildContext(ctx, maxSyncMatchWaitTime, 0) // wait for a poller for maxSyncMatchWaitTime before forwarding to another partition or writing to database
 	select {
 	case tm.getTaskC(task) <- task: // poller picked up the task
+		cancel()
 		if task.ResponseC != nil {
 			// if there is a response channel, block until resp is received
 			// and return error if the response contains error
 			err := <-task.ResponseC
 			return true, err
 		}
 		return false, nil
-	default:
+	case <-childCtx.Done():
```

Comment:
Please hold onto this change until I have the matching simulation test ready. I was playing with this and a few other `select {}` blocks that does similar stuff. The outcome is unclear because I need to measure more things probably.